components/net_traits/bluetooth_blacklist.rs

```diff
@@ -0,0 +1,120 @@
+
+pub struct BluetoothBlacklist {
+    blacklist: HashMap<String, ValueType>,
+    error_string: String,
+}
+
+#[derive(Eq, PartialEq)]
+pub enum ValueType {
+    Exclude,
+    ExcludeReads,
+    ExcludeWrites,
+}
+
+impl BluetoothBlacklist {
+    pub fn new() -> BluetoothBlacklist {
+        BluetoothBlacklist {
+            blacklist: HashMap::new(),
+            error_string: String::new(),
+        }
+    }
+
+    pub fn new_initialized() -> BluetoothBlacklist {
+        let mut btblist = BluetoothBlacklist::new();
+        if let Err(error) = btblist.fill_blacklist() {
+            btblist.error_string = error;
+        }
+        btblist
+    }
+
+    pub fn is_excluded(&mut self, uuid: &str) -> Result<(), String> {
+        if !self.error_string.is_empty() {
+            return Err(self.error_string.clone());
+        }
+        if self.blacklist.get(uuid).map_or(false, |value| value.eq(&ValueType::Exclude)) {
+            return Err(String::from(EXCLUDE_ERR));
+        }
+        Ok(())
+    }
+
+    pub fn is_excluded_from_reads(&mut self, uuid: &str) -> Result<(), String> {
+        try!(self.is_excluded(uuid));
+        if self.blacklist.get(uuid).map_or(false, |value| value.eq(&ValueType::ExcludeReads)) {
+            return Err(String::from(EXCLUDE_READS_ERR));
+        }
+        Ok(())
+    }
+
+    pub fn is_excluded_from_writes(&mut self, uuid: &str) -> Result<(), String> {
+        try!(self.is_excluded(uuid));
+        if self.blacklist.get(uuid).map_or(false, |value| value.eq(&ValueType::ExcludeWrites)) {
+            return Err(String::from(EXCLUDE_WRITES_ERR));
+        }
+        Ok(())
+    }
+
+    pub fn fill_blacklist(&mut self) -> Result<(), String> {
+        let path = Path::new(FILE_PATH);
+        let mut file = match File::open(&path) {
+            Ok(file) => file,
+            Err(_) => return Err(format!("Couldn't open {}", path.display())),
+        };
+        let mut data = String::new();
+        if file.read_to_string(&mut data).is_err() {
+            return Err(String::from(BLACKLIST_PARSE_ERROR));
+        }
+        for line in data.lines() {
+            let linestr = line.to_string();
+            if linestr.is_empty() || linestr.starts_with('#') {
+                continue;
+            } else {
+                let mut value = ValueType::Exclude;
+                let mut words = line.split_whitespace();
+                if let Some(uuid) = words.next() {
+                    if let Some(second) = words.next() {
+                        if second.eq(EXCLUDE_READS) {
+                            value = ValueType::ExcludeReads;
+                        }
+                        if second.eq(EXCLUDE_WRITES) {
+                            value  = ValueType::ExcludeWrites;
+                        }
+                    }
+                    if !self.validate_uuid(uuid) {
+                        self.blacklist.clear();
```

Comment:
Because if there's an error in the begining of the file and we keep the blacklist as it is, we could connect to a UUID which is blacklisted later in the file (i.e. in the end of the file).